heidrun/pa_dc.rb

```diff
@@ -0,0 +1,108 @@
+
+# Convert ValueArray to an array to manipulate and then reinstantiate a new ValueArray
+#   If more than one dc:contributor property exists then it drops the last one 
+#   If one or zero exist then do nothing
+# 
+# Feels kludgey as heck, should rewrite once ValueArray supports array sliceing 
+# See ticket https://issues.dp.la/issues/8287
+contributor_top = lambda do |r|
+  # testing 
+  # r['dc:contributor'].first_value( r['dc:contributor'].count-1 )
+  # end test
+
+  if r['dc:contributor'].count > 1 
+    r['dc:contributor'].first_value( r['dc:contributor'].count-1 )
+  else 
+    r['dc:contributor']
+  end 
+end
+
+# dc:relation 
+# If there is more than one dc:relation property than return all but the first 
+# instance of the property. If one or none then return an empty set
+#
+secondary_relations = lambda do |r|
+  if(r['dc:relation'].count > 1)
+    r['dc:relation'].last_value( r['dc:relation'].count-1 )
+  else 
+    # TODO there is probably a better way return an empty set that isn't nil
+    r['dc:relation'].last_value(0)
+  end 
+end 
+
+
+Krikri::Mapper.define(:pa_dc, :parser => Krikri::OaiDcParser) do
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/pennsylvania'
+    label 'Pennsylvania Digital Collections Project'
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+    # TODO -- dbl check
+    collection record.field('dc:relation').first_value
+
+    contributor record.map( &contributor_top ).flatten
+
+    creator :class => DPLA::MAP::Agent, :each => record.field('dc:creator'), :as => :creator do
+      providedLabel creator
+    end
+
+    date :class => DPLA::MAP::TimeSpan, :each => record.field('dc:coverage'), :as => :coverage do
+      providedLabel coverage
+    end
+
+    description record.field('dc:description')
+
+    # Selecting non-DCMIType values will be handled in enrichment <<-- CONFIRM 
```

Comment:
@no-reply does that change the mapping or are you just identifying the enrichment segment responsible for dropping non-DCMI types?